torch/csrc/autograd/custom_function.cpp

```diff
@@ -0,0 +1,84 @@
+#include <torch/csrc/autograd/custom_function.h>
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+
+namespace torch { namespace autograd {
+
+std::vector<Variable> _wrap_outputs(const std::unordered_set<at::TensorImpl*> &inputs,
+  const std::unordered_set<at::TensorImpl*> &non_differentiable,
+  const std::unordered_set<at::TensorImpl*> &dirty_inputs,
+  const std::vector<Variable> &raw_outputs,
```

Comment:
C++ in our codebase nit: it's more idiomatic to take an `ArrayRef<Variable> raw_outputs` here. ArrayRef is our general purpose "reference to vector-like struct"; it can take out a reference to vectors, but other formats too.